### PR TITLE
Add Kusama / Polkadot address

### PR DESCRIPTION
**Pull request by hewigovens**
## Description

Decide to create separated pull request, don't want to always rebase / merge master :( `sr25519` is not mandatory, we will keep using `ed25519` here
Note the derivation path and its address is not compatible with `subkey`

## Testing instructions

Run UT

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: src/HDWallet.h [hunk 1]
```diff
@@ -58,6 +58,9 @@ class HDWallet {
 
     virtual ~HDWallet();
 
+    /// Returns master key.
+    PrivateKey getMasterKey(TWCurve curve) const;
```
Comment (catenocrypt) [inline src/HDWallet.h]: Why is masterKey needed?  I don't see it used.

File: src/SS58Address.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+#include <array>
+#include <string>
+#include <iostream>
+
+const std::string SS58Prefix = "SS58PRE";
+
+namespace TW {
+
+class SS58Address {
```
Comment (catenocrypt) [inline src/SS58Address.h]: What is 'SS58'stands for?  Could you add a comment?

Comment (hewigovens) [reply src/SS58Address.h]: SS58 is short for **S**ub**S**trate Base**58** checksum, we have a link in `TWSS58AddressType.h`

File: src/HDWallet.h (hunk 1 above)
Comment (hewigovens) [reply src/HDWallet.h]: not used but useful, with this I find that subkey is not compatible with slip-0010

**Comment by hewigovens** [comment]
> Naming: So the blockchain is Polkadot, but the coin is Kusama. I wonder whether the address scheme is also Kusama-specific?

Kusama runs same code as Polkadot (can be treated as a testnet), address scheme is same as SS58 (Substrate based)

File: include/TrustWalletCore/TWHDWallet.h [hunk 3]
```diff
@@ -49,6 +49,10 @@ TWData *_Nonnull TWHDWalletSeed(struct TWHDWallet *_Nonnull wallet);
 TW_EXPORT_PROPERTY
 TWString *_Nonnull TWHDWalletMnemonic(struct TWHDWallet *_Nonnull wallet);
 
+/// Returns master key.
+TW_EXPORT_METHOD
+struct TWPrivateKey *_Nonnull TWHDWalletGetMasterKey(struct TWHDWallet *_Nonnull wallet, enum TWCurve curve);
```
Comment (vikmeup) [inline include/TrustWalletCore/TWHDWallet.h]: is master key needed?

Comment (hewigovens) [reply include/TrustWalletCore/TWHDWallet.h]: see: https://github.com/trustwallet/wallet-core/pull/754#discussion_r354062662

**Pull request merged**